submissions/VBystrov/friends-app/js/index.js

```diff
@@ -0,0 +1,328 @@
+  }
+
+  init(url) {
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((friendsData) => {
+        const friendlist = new Friendlist(friendsData.results);
+        const search = new Search(friendlist);
+      })
+      .catch((error) => {
+        console.error(
+          'There has been a problem with your fetch operation:',
+          error
+        );
+      });
+  }
+}
+
+class Friendlist {
+  constructor(friendsData) {
+    this.container = document.getElementById('friendlist');
+    this.createCards(friendsData);
+    this.sorters = new Sorters();
+  }
+
+  createCards(friendlistData) {
+    this.cards = friendlistData.map((cardData) => new Card(cardData));
+  }
+
+  applySearch(searchParameters) {
+    this.applyFilters(searchParameters);
+    this.applySorting(searchParameters.sorting);
+  }
+
+  applyFilters(searchParameters) {
+    this.filteredCards = [...this.cards];
+
+    if (searchParameters.age) {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.age === searchParameters.age
+      );
+    }
+
+    if (searchParameters.name) {
+      this.filteredCards = this.filteredCards.filter((card) =>
+        `${card.firstName} ${card.lastName}`
+          .toLowerCase()
+          .includes(searchParameters.name.toLowerCase())
+      );
+    }
+
+    if (searchParameters.gender !== 'all') {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.gender === searchParameters.gender
+      );
+    }
+  }
+
+  applySorting(sortingOrder) {
+    this.sortedCards = [...this.filteredCards].sort(this.sorters[sortingOrder]);
+  }
+
+  showCards() {
+    if (this.sortedCards.length) {
+      this.container.innerHTML = this.sortedCards
+        .map(({ html }) => html)
+        .join('');
+    } else {
+      this.container.innerHTML = `<div class="not-found">
+        <p>No users found on request.</p>
+      </div>`;
+    }
+  }
+}
+
+class Search {
+  constructor(friendlist) {
+    this.container = document.getElementById('search');
+    this.inputName = this.container.querySelector('#search-name__text');
+    this.inputAge = this.container.querySelector('#search-age__number');
+    this.genderContainer = this.container.querySelector('#search-gender');
+    this.friendlist = friendlist;
+    this.parameters = this.parseUrl();
+    this.setInitialSearch(this.parameters);
+    this.friendlist.applySearch(this.parameters);
+    this.friendlist.showCards();
+
+    this.container.addEventListener('click', this.changeSorting.bind(this));
+    this.inputName.addEventListener('input', this.changeSearchName.bind(this));
+    this.inputAge.addEventListener('input', this.changeSearchAge.bind(this));
+    this.genderContainer.addEventListener(
+      'click',
+      this.changeSearchGender.bind(this)
+    );
+  }
+
+  parseUrl() {
```

Comment:
Move URL-related stuff into a separate entity. I don't think that `Search` should be responsible for handling interactions with URL